instance/tests/models/factories/openedx_appserver.py

```diff
@@ -49,4 +50,68 @@ def make_test_appserver(instance=None, s3=False, server=None):
         appserver.server = server
         appserver.save()
 
+    if status == AppServerStatus.Running:
+        _set_appserver_running(appserver)
+    if status == AppServerStatus.ConfigurationFailed:
+        _set_appserver_configuration_failed(appserver)
+    elif status == AppServerStatus.Error:
+        _set_appserver_errored(appserver)
+    elif status == AppServerStatus.Terminated:
+        _set_appserver_terminated(appserver)
+
     return appserver
+
+
+def _set_appserver_terminated(appserver):
+    """
+    Transition `appserver` to AppServerStatus.Terminated.
+    """
+    appserver._status_to_waiting_for_server()
+    appserver._status_to_configuring_server()
+    appserver._status_to_running()
+    appserver._status_to_terminated()
+
+
+def _set_appserver_running(appserver):
+    """
+    Transition `appserver` to AppServerStatus.Running.
+    """
+    appserver._status_to_waiting_for_server()
+    appserver._status_to_configuring_server()
+    appserver._status_to_running()
+
+
+def _set_appserver_configuration_failed(appserver):
+    """
+    Transition `appserver` to AppServerStatus.ConfigurationFailed.
+    """
+    appserver._status_to_waiting_for_server()
+    appserver._status_to_configuring_server()
+    appserver._status_to_configuration_failed()
+
+
+def _set_appserver_errored(appserver):
+    """
+    Transition `appserver` to AppServerStatus.Error.
+    """
+    appserver._status_to_waiting_for_server()
+    appserver._status_to_error()
+
+
+def _set_server_ready(server):
```

Comment:
Noone is using this function so I think it should be removed. It's probably not even about AppServer but maybe about Server.